solutions/instances/main.tf

```diff
@@ -147,11 +159,57 @@ locals {
   log_analysis_instance_name     = var.prefix != null ? "${var.prefix}-${var.log_analysis_instance_name}" : var.log_analysis_instance_name
   cloud_monitoring_instance_name = var.prefix != null ? "${var.prefix}-${var.cloud_monitoring_instance_name}" : var.cloud_monitoring_instance_name
   cloud_logs_instance_name       = var.prefix != null ? "${var.prefix}-cloud-logs" : var.cloud_logs_instance_name
+  cloud_logs_data_bucket_crn     = var.existing_cloud_logs_data_bucket_crn != null ? var.existing_cloud_logs_data_bucket_crn : module.cos_bucket[0].buckets[local.cloud_log_data_bucket].bucket_crn
+  cloud_log_metrics_bucket_crn   = var.existing_cloud_logs_metrics_bucket_crn != null ? var.existing_cloud_logs_metrics_bucket_crn : module.cos_bucket[0].buckets[local.cloud_log_metrics_bucket].bucket_crn
+  cloud_logs_buckets             = [local.cloud_logs_data_bucket_crn, local.cloud_log_metrics_bucket_crn]
+}
+
+data "ibm_iam_account_settings" "iam_account_settings" {
```

Comment:
shouldn't this data lookup be done with the cos alias provider? Otherwise its going return the account that observability instances are in. But we seem to already have that one line 366 so why are we doing another one here?

Reply:
This one is for current account , on line 366 that is for cos account
<img width="639" alt="Screenshot 2024-10-10 at 8 02 37 PM" src="https://github.com/user-attachments/assets/ee6a7f20-56cd-49a5-89f0-87469f19e3cb">

Reply:
@Khuzaima05 but you are using this data lookup on line 187. But line 187 should be the COS account. So this breaks in the cross account use case. Did you not hit this when testing?

Reply:
@ocofaigh , This auth policy is getting created only when we are using cross-account.
We have put a validation on line 172 for same.

Reply:
I think you are missing the point. The data lookup on line 167 is always going to return the account ID that Observability resources are getting created.
You are using this account ID on line 187. So it means your are NOT adding the correct account ID in that auth policy. The account ID should be from the COS account, not account Observability is in

Reply:
@ocofaigh I have updated the code, if you check line number 173 that the source account id, this where our observability instance will be created.
and on line number 187 we are passing the account id for the cos instance, passing the account id is required when creating a narrowed down auth policy for logs to access only the bucket.

Reply:
ah ok line 187 is not using that data lookup - the resource name is so similar - I think we are good